drivers/docker/handle.go

```diff
@@ -212,15 +212,6 @@ func (h *taskHandle) run() {
 		}
 	}
 
-	// Remove the container
-	if h.removeContainerOnExit == true {
-		if err := h.client.RemoveContainer(docker.RemoveContainerOptions{ID: h.containerID, RemoveVolumes: true, Force: true}); err != nil {
-			h.logger.Error("error removing container", "error", err)
-		}
-	} else {
-		h.logger.Debug("not removing container due to config")
-	}
-
```

Comment:
Container removed was moved from here, in `run`, to `DestroyTask`: 
https://github.com/hashicorp/nomad/pull/5768/files#diff-2b17483fc1838ce536b7f8e6f6f6ea8eR1122